### ISSUE TITLE
One or more commits after tag: asio-1-12-2 caused compilation failure on Cygwin and Mingw

### ISSUE THREAD

**Issue by youngwolf-project** [boostorg/asio#206]
Before, standalone asio can be compiled on 32bit Cygwin, 32 and 64bit Mingw.
Now it cannot be compiled on all Cygwin and Mingw.

**Comment by tanzislam** [comment]
@yongwolf-project what are the compilation errors that you're facing? I'm facing these:

```
C:/Extension/Programs/Programming/boost/boost/asio/windows/basic_random_access_handle.hpp:93:13: error: 'native_handle_type' does not name a type
       const native_handle_type& handle)
             ^~~~~~~~~~~~~~~~~~
C:/Extension/Programs/Programming/boost/boost/asio/windows/basic_random_access_handle.hpp:93:13: note: (perhaps 'typename boost::asio::windows::basic_overlapped_handle<Executor>::native_handle_type' was intended)
C:/Extension/Programs/Programming/boost/boost/asio/windows/basic_random_access_handle.hpp:113:13: error: 'native_handle_type' does not name a type
       const native_handle_type& handle,
             ^~~~~~~~~~~~~~~~~~
C:/Extension/Programs/Programming/boost/boost/asio/windows/basic_random_access_handle.hpp:113:13: note: (perhaps 'typename boost::asio::windows::basic_overlapped_handle<Executor>::native_handle_type' was intended)
```

**Comment by youngwolf-project** [comment]
@tanzislam 
Exactly.

**Comment by tanzislam** [comment]
This was fixed in 483f3d3 in "develop", though yet to reach the "master" branch of this project and the superproject.

**Comment by youngwolf-project** [comment]
@tanzislam Thanks, it's already available.
